### PR TITLE
chore(GcpNfsVolumeRestore): Unification of Status and State

### PR DESCRIPTION
**Pull request by ravi-shankar-sap**
**Description**

Changes proposed in this pull request:

- Unification of State and Status for GcpNfsVolumeRestore

**Related issue(s)**
https://jira.tools.sap/browse/PHX-127

### CODE REVIEW

File: pkg/skr/gcpnfsvolumerestore/checkStates.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package gcpnfsvolumerestore
```
Comment (abalaie) [inline pkg/skr/gcpnfsvolumerestore/checkStates.go:1]: Minor: checkStates is very vague and doesn't really explain what this action do. This is just to set the Processing state if it was in error (which is not a final) state.

Comment (ravi-shankar-sap) [reply pkg/skr/gcpnfsvolumerestore/checkStates.go]: Changed the name to `setProcessing`.

File: pkg/skr/gcpnfsvolumerestore/checkStates.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	// If the status is FAILED or DONE, i.e. completed, skip
+	if restore.Status.State == cloudresourcesv1beta1.JobStateFailed ||
+		restore.Status.State == cloudresourcesv1beta1.JobStateDone {
+		return composed.StopAndForget, nil
+	}
+
+	//If state is empty or error, Reset the error conditions, and restart with processing state
+	if restore.Status.State == "" || restore.Status.State == cloudresourcesv1beta1.JobStateError {
+		restore.Status.State = cloudresourcesv1beta1.JobStateProcessing
```
[the remark this answers is not in the document]
Comment (ravi-shankar-sap) [reply pkg/skr/gcpnfsvolumerestore/checkStates.go]: Updated the condition now to set it to `processing` only at the beginning.

File: pkg/skr/gcpnfsvolumerestore/setProcessing.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+	logger.WithValues("NfsRestore :", restore.Name).Info("Checking States")
+
+	// If the status is FAILED or DONE, i.e. completed, skip
+	if restore.Status.State == cloudresourcesv1beta1.JobStateFailed ||
+		restore.Status.State == cloudresourcesv1beta1.JobStateDone {
+		return composed.StopAndForget, nil
+	}
+
+	//If state is empty or error, Reset the error conditions, and restart with processing state
```
Comment (abalaie) [inline pkg/skr/gcpnfsvolumerestore/setProcessing.go:28]:
```suggestion
	//If state is empty, Reset the error conditions, and restart with processing state
```

**Review by abalaie — APPROVED** [review]
lgtm

**Pull request merged**